server/controllers/profile.js

```diff
@@ -71,5 +71,44 @@ exports.uploadProfileImage = asyncHandler(async (req, res, next) => {
       .status(200)
       .send({ message: "The profile has been successfully updated" });
   });
+});
+
+// @route POST /profile/edit-availability
+// @desc edit a users availability
+// @access Private
+exports.editAvailability = asyncHandler(async (req, res, next) => {
+  const profile = await User.findById(req.user.id, "profile");
+
+  if (!profile) {
+    res.status(401);
+    throw new Error("Not authorized");
+  }
+  const timeSlot = await Profile.findOneAndUpdate(
+    { _id: profile, "availability.day": req.body.date.date },
```

Comment:
```suggestion
    { _id: profile, "availability.day": req.body.selectedDate.date },
```
Suggestion: You can change the name of the input parameter to avoid any confusion :)